### DISCUSSION TITLE
Led Strip

### DISCUSSION THREAD

**Discussion by duvamduvam** [Pi4J/pi4j#169] [Q&A] [answered]
Hi,
I m currently working on a street theater robot, I have a version working on arduino, but I would like to switch for raspberry with java to go more far with programming.
The robot have wheels, servo, leds strips and led matrix.
I see how it could work for wheels and servo, but I don't for leds strips.
Is it possible to do with pi4j something like : void updateLed(Pwm, pwm, int ledNb, Color color) ?
Thanks a lot for the help.

David

**Comment by FDelporte** [comment]
To be honest, controlling LED strips with Java is a new path for me too. I checked the Adafruit implementation with an SPI strip and this should be pretty feasible to achieve in Java... 
https://github.com/adafruit/Adafruit_CircuitPython_WS2801/blob/main/adafruit_ws2801.py

So far, I've been only experimenting with WS2812-strips with a one-wire connection + Python, but I'm definitely going to order a WS2801-strip and see what can be achieved with SPI (if time is available).

**Comment by duvamduvam** [comment]
Thanks for the reply, I ll do python connectors then.
I need to use pwm as I need several matrix and strip.

I found this post here mentioning pi4j : https://forums.raspberrypi.com/viewtopic.php?t=187520
It was then possible with the previous API ?

**Comment by FDelporte** [comment, marked as answer]
FYI a Raspberry Pi is not ideal to control (long) strips anyhow because of timing-critical data.
Maybe consider to combine a Java app with an extra microcontroller-kind of device for the LED strip control.

In my book ["Getting Started with Java on the Raspberry Pi"](https://webtechie.be/books/) I did a similar thing with an Arduino, sources on https://github.com/FDelporte/JavaOnRaspberryPi/tree/master/Chapter_11_Queues/arduino-wifi-mosquitto

And a few other examples after a quick search
* https://github.com/vadimtitov/esp32_rgb_server
* https://ledstripstudio.com/ (pro products)

**Comment by FDelporte** [comment]
And I was actually searching for this project, but only found now: https://www.bhencke.com/pixelblaze

```
Pixelblaze V2+ is an advanced WiFi LED controller and pattern development engine.
...
Pixelblaze has WiFi built-in and serves up a web page with a pattern list and pattern editor. Changes in the editor are compiled on the fly and updated in Pixelblaze so you can see your changes live.
...
up to 5,000 APA102 LED or 2500 WS2812 LEDs
...
Pixelblaze V3 and V3 Pico was designed to be embeddable and integrated into LED art pieces or used in costumes and props. 
```

**Comment by duvamduvam** [comment]
Thanks for checking Frank, but I have already this working remotly via lora radio system on arduino
I just added some photo so you can see what it looks like : https://duvam.net/didier/
It's a street theater robot, I can change light do face expression, move the head and the wheels.
But I am a lousy C dev, I wanted to go more far on the progamming, for instance make the robot dancing by making synchronized sequences of light and move and expression.
I ll try to see how far I can go with python / java , I started already https://github.com/duvamduvam/didier-java
Wish me luck ;)

**Comment by FDelporte** [comment]
@duvamduvam took some time, but finally added to featured projects: https://pi4j.com/featured-projects/street-artist-robot/
Please feel free to provide more text and info that we can add to this page!
